build/VSUtilities.psm1

```diff
@@ -0,0 +1,119 @@
+function Start-VsDevShell {
+    $chosenVisualStudioVersion = ChooseVisualStudioInstallation @(Get-VSInstalls)
+    if ($chosenVisualStudioVersion -eq $null) {
+            return
+    }
+
+    $installDir = $chosenVisualStudioVersion.InstallationPath
+    $devShellModule = Join-Path $installDir 'Common7\Tools\Microsoft.VisualStudio.DevShell.dll'
+    Import-Module $devShellModule
+    Enter-VsDevShell -DevCmdArguments "-no_logo" -StartInPath "$pwd" $chosenVisualStudioVersion.InstanceId
+}
+
+function Get-VsCommonToolsPath {
+    if([string]::IsNullOrEmpty($env:VSINSTALLDIR)) {
+        $chosenVisualStudioVersion = ChooseVisualStudioInstallation @(Get-VSInstalls)
+        $installDir = $chosenVisualStudioVersion.InstallationPath
+    }
+    else {
+        $installDir = $env:VSINSTALLDIR
+    }
+
+    if([string]::IsNullOrEmpty($installDir)) {
+        return
+    }
+
+    $vsComnToolsPath = Join-Path $installDir 'Common7\Tools\'
+    return $vsComnToolsPath
+}
+
+function Get-VSWhere {
+    $ProgramFile = ${env:ProgramFiles(x86)}
+
+    if ($null -eq $ProgramFile) {
+        $ProgramFile = (Get-ChildItem env:ProgramFiles).Value
+    }
+
+    $vsWhereVersion = "2.8.4"
+    $vsWhereDir = Join-Path $ProgramFile 'Microsoft Visual Studio\Installer'
+    $vsWhereExe = "$vsWhereDir\vswhere.exe"
+    
+    if (!(Test-Path $vsWhereExe)) {
+        if (!(Test-Path $vsWhereDir)) {
+            Write-Host "Creating folder $vsWhereDir"
+            New-Item -ItemType Directory -Path $vsWhereDir
+        }
+
+        Write-Host "Downloading vswhere"
+        [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+        Invoke-WebRequest "http://github.com/Microsoft/vswhere/releases/download/$vsWhereVersion/vswhere.exe" -OutFile $vswhereExe
```

Comment:
Firstly, this looks like it's trying to create a directory and write a file in program files, which I strongly believe we should not be doing.

Secondly, I believe vswhere first shipped with VS2017, 15.3. Given the current NuGet dev branch needs VS2019, it's safe to assume that the machine has vswhere installed, so no need to download. THe VS installer doesn't allow customers to modify where the installer, and therefore where vswhere, is saved, so I think it's safe enough to hardcode the program files location and if it's not found, assume no compatible VS is installed.  `build\common.ps1` also uses vswhere.